### ISSUE TITLE
How much scalable is the client?

### ISSUE THREAD

**Issue by Leonardo-Ferreira** [rabbitmq/rabbitmq-dotnet-client#498]
Looking at the code, on several places I saw `lock`'s statements. This issued an internal question on how much does that impact performance.

My current solution is a web-app that is using graylog for logging, and its sink is a rabbit queue. A single critical path request can result on several dozens of logs alone, and ideally it should run on 500ms. On peak moments we´re expecting to handle 3-5 of those requests and 1-2 hundreds of others each second.

Right now, the connection and the model are basically singletons and my question is: how worried should i be about those locks when we hit heavy load? are there know deadlocks spots?

**Comment by lukebakken** [comment]
Thanks for using this client library and (presumably) RabbitMQ.

When you click "New Issue" in this repository, the following text pre-fills the comment box:

----

## Does This Belong to GitHub or RabbitMQ Mailing List?

*Important:* please first read the `CONTRIBUTING.md` document in the
root of this repository. It will help you determine whether your
feedback should be directed to the RabbitMQ mailing list [1] instead.

----

Please see [this section](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/master/CONTRIBUTING.md#github-issues) of `CONTRIBUTING.md` which explains why GitHub issues are not used for questions.

Thanks

**Comment by michaelklishin** [comment]
The only right answer is "do your own benchmarking and profiling". This client has never blocked on the hot consumer patch IIRC and these days has a separate "particularly asynchronous" consumer dispatch implementation as an option. As long as the [single giant queue anti-pattern](http://www.rabbitmq.com/queues.html#runtime-characteristics) is not used, 300 messages/second is a rate any client can very easily cope with.
